### DISCUSSION TITLE
ALTAYA animals of the farm

### DISCUSSION THREAD

**Discussion by maximedub** [oyooyo/audiocube#19] [General]
Hello, I bought the audio box of animals of the farms in French. I have the smp files. I try to decrypt alls files but encryption is different. What is the process to find encryption of smp files ?

**Comment by oyooyo** [comment]
Have you tried decrypting the files with the algorithm of a different device? As far as I remember, someone told me that they are now selling a new device somewhere that is blue, but uses the encryption of the green device, or the other way around.

If that doesn't work: A good way to find out the encryption is to look at the end of the SMP files, where MP3 metadata is stored. One should be able to see a repeating sequence of a few bytes there, and it should be possible to extract the encryption key from this byte sequence.

**Comment by maximedub** [comment]
Hi, you speak about this kind of byte sequence : 

> 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 
DD EA 66 F7 D3 12 43 9F 5D 0B 03 97 E7 D1 8E BE C6 59 85 BB DA 44 11 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 
08 72 6F 54 77 1D 1B 28 6A 06 77 44 11 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 
CC D9 33 80 C3 22 14 1C 6E 1B B5 C0 74 42 A8 49 9B D6 25 CE F5 77 44 11 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 
66 77 5D 05 7E 67 22 6A 0A 1B 3F 71 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 
66 77 44 BB C8 02 D5 BE B3 24 39 1B 78 84 1D 87 C2 7F 8E 26 EB D4 04 39 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 
66 DD EA 66 F7 CE 92 43 77 87 13 00 B7 46 19 F4 3D 35 19 54 3E 72 44 11 66 77 44 11 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 
CC D9 31 80 FA 21 D1 44 32 86 11 44 33 02 11 44 07 A2 11 44 37 77 44 11 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 
66 77 EE BF 11 E6 D9 C7 F3 22 10 E0 33 22 11 64 33 22 25 C4 33 22 15 11 
66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66 77 44 11 66

**Reply by oyooyo** [reply]
Yes, exactly. The repeating pattern looks like it's using the same or least very similar encryption as the devices I've already seen, with a 4 byte key, but maybe a different key is used. But I would probably need to have 1 or 2 complete .SMP files to fully figure the encryption out.

**Comment by maximedub** [comment]
Hi, sorry for not respond for a long time. You can find files here :
https://1drv.ms/u/s!Ar9FSsMzQ95wkpNJtczvUjAjwpZkuw?e=b37AP2

**Comment by oyooyo** [comment]
Thanks, I analyzed the files and it turned out that this is indeed a new type of device/encryption that was not supported by audiocube so far.
It uses a simple XOR-encryption with the key [0x44, 0x33, 0x22, 0x11].
I've just pushed a new version of the audiocube.py file that has basic support for this device, meaning you can encrypt/decrypt .smp files.
For example, a command like
`audiocube.py altaya decrypt MV002.smp`
would decrypt file MV002.smp to MV002.mp3
or
`audiocube.py altaya encrypt MV001.mp3`
would encrypt file MV001.mp3 to MV001.smp

It would be good to have a photo of the device with a figure, as the other devices have. It would also be good to know more about the NFC tags in the base of the figures so that it's possible to make your own NFC tags/figures.
